src/unstakeAg/index.ts

```diff
@@ -799,20 +835,53 @@ export class UnstakeAg {
       user,
       feeAccounts,
       assumeAtasExist,
+      asLegacyTransaction,
     });
-    if (!exchangeReturn.setupTransaction) {
-      exchangeReturn.setupTransaction = {
+
+    if (asLegacyTransaction) {
+      const legacyExchangeReturn = exchangeReturn as ExchangeReturn;
+      legacyExchangeReturn.setupTransaction = {
         tx: new Transaction(),
         signers: [],
       };
```

Comment:
does this assignment overwrite the existing `legacyExchangeReturn.setupTransaction`?